packages/react-native/React/Base/RCTUtils.mm

```diff
@@ -383,21 +383,27 @@ CGFloat RCTFontSizeMultiplier(void)
   return mapping[RCTSharedApplication().preferredContentSizeCategory].floatValue;
 }
 
+UIDeviceOrientation RCTDeviceOrientation(void) {
+  return [[UIDevice currentDevice] orientation];
+}
+
 CGSize RCTScreenSize(void)
 {
-  // FIXME: this caches whatever the bounds were when it was first called, and then
-  // doesn't update when the device is rotated. We need to find another thread-
-  // safe way to get the screen size.
```

Comment:
Does screen size change if we have something insetting the window? Like, a keyboard open, or if a window on an iPad is resized?

I agree with the comment, that this is pretty sketchy 😀